packages/relay/tests/lib/subscriptionController.spec.ts

```diff
@@ -0,0 +1,221 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+
+import pino from "pino";
+// @ts-ignore
+import {SubscriptionController} from "../../src/lib/subscriptionController";
+import {expect} from "chai";
+import {Poller} from "../../src/lib/poller";
+import {EthImpl} from "../../src/lib/eth";
+import {MirrorNodeClient, SDKClient} from "../../src/lib/clients";
+import MockAdapter from "axios-mock-adapter";
+import LRU from "lru-cache";
+import constants from "../../src/lib/constants";
+import sinon from 'sinon';
+import {Registry} from 'prom-client';
+import dotenv from "dotenv";
+import path from 'path';
+
+dotenv.config({ path: path.resolve(__dirname, '../test.env') });
+
+const logger = pino();
+const registry = new Registry();
+
+let restMock: MockAdapter;
+let mirrorNodeInstance: MirrorNodeClient;
+let sdkClientStub;
+let cache;
+let ethImpl: EthImpl;
+let poller: Poller;
+
+class MockWsConnection {
+
+    id: string;
+
+    constructor(id: string) {
+        this.id = id;
+    }
+
+    send(msg) {
+        console.log(`Mocked ws-connection with id: ${this.id} used method: send(${msg}`);
+    }
+
+}
+
+describe("subscriptionController", async function() {
+   this.timeout(20000);
+   let subscriptionController: SubscriptionController;
+   let sandbox;
+
+   this.beforeAll(() => {
+       // @ts-ignore
+       mirrorNodeInstance = new MirrorNodeClient(process.env.MIRROR_NODE_URL, logger.child({ name: `mirror-node` }), registry);
+       // @ts-ignore
+       restMock = new MockAdapter(mirrorNodeInstance.getMirrorNodeRestInstance(), { onNoMatch: "throwException" });
+
+       sdkClientStub = sinon.createStubInstance(SDKClient);
+       cache = new LRU({
+           max: constants.CACHE_MAX,
+           ttl: constants.CACHE_TTL.ONE_HOUR
+       });
+       // @ts-ignore
+       ethImpl = new EthImpl(sdkClientStub, mirrorNodeInstance, logger, '0x12a', cache);
+       poller = new Poller(ethImpl, logger);
+
+       subscriptionController = new SubscriptionController(poller, logger);
+   });
+
+   this.beforeEach(()=> {
+       cache.clear();
+       restMock.reset();
+       sandbox = sinon.createSandbox();
+   });
+
+    this.afterEach(() => {
+        restMock.resetHandlers();
+        sandbox.restore();
+    });
+
+    it('Should create sha256 hash out of a data object', async function () {
+        const dataToHash = "This is a Test";
+
+        const hash = subscriptionController.createHash(dataToHash);
+
+        expect(hash).to.be.eq(`401b022b962452749726ba96d436921e39d6deb2b0f4a922cc3da5d7e99e6e46`);
+    });
+
+    it('generateId should create a random hex ID, with 34 length', async function () {
+        const generatedId = subscriptionController.generateId();
+
+        expect(generatedId).to.be.length(34);
```

Comment:
Add check for `0x` prefix.
The presence and lack has come up in the past so we should be explicit in our tests on which one it is

Reply:
✅ Done